### PR TITLE
Rewrite js2 week 2 prep content

### PR DESCRIPTION
**Pull request by Dedekind561**
## What does this change?

Module: JS2
Week(s): 2

## Checklist

- [x] I have read the [contributing guidelines](CONTRIBUTING.MD)
- [x] I have checked my spelling and grammar with an [automated tool](https://www.grammarly.com/grammar-check)
- [x] I have previewed my changes to check the [markdown renders](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) as I intend
- [x] I have run my code to check it works
- [ ] My changes follow our [Style Guide](https://curriculum.codeyourfuture.io/guides/code-style-guide)

### CODE REVIEW

**Review by illicitonion — COMMENTED** [review]
Looking great!

See also https://github.com/CodeYourFuture/curriculum/pull/273

**Comment by illicitonion** [comment]
Oh also, do we want to explicitly have an example of "an object containing an array" and "an array containing an object" somewhere, to show composition?

File: content/js2/blocks/no-params/index.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+```console
+If it should pass with deep equality, replace "toBe" with "toStrictEqual"
+```
+
+In other words, `toBe` checks the **references** of the two objects. As `currentOutput`and `targetOutput` point to different objects - this can never be true. However, we can use a different matcher that compares the **contents** of the two objects. We can use [`toStrictEqual`](https://jestjs.io/docs/expect#tostrictequalvalue) to check that both objects have exactly the same contents:
+
+```js
+expect(currentOutput).toStrictEqual(targetOutput);
+```
```
[the remark this answers is not in the document]
Comment (Dedekind561) [reply content/js2/blocks/no-params/index.md:52]: Yeah, we should get them to research a definition of the term "serialize" as you said and then use that definition to interpret the error message. Definitely a challenging but worthwhile exercise. I think we should ask them to post in Slack and see if anybody can improve on somebody's explanation in a Slack thread or something.

[the remark this answers is not in the document]
Comment (Dedekind561) [reply content/js2/blocks/no-params/index.md:52]: I love explanation evolution as an activity in general

File: content/js2/blocks/query-string/index.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+_Given_ a query string and a function `parseQueryString`,
+_When_ we call `parseQueryString` with a query string,
+_Then_ it should return an object with the key-value pairs
+
+E.g.
+
+```js
+parseQueryString("colour=blue&sort=newest");
+// should return { colour: "blue, sort: "newest" }`
+```
```
[the remark this answers is not in the document]
Comment (Dedekind561) [reply content/js2/blocks/query-string/index.md:34]: I think these points are very interesting and definitely worth exploring in greater depth. But I think the prep content would be too large if we added them here. I think they could go in a compare-and-interpret / compare-and-contrast style exercise in the coursework though. The exercise

File: content/js2/blocks/query-string/index.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
++++
+title = '❓🪢 Query strings'
+headless = true
+time = 10
+facilitation = false
+emoji= '🧩'
+[objectives]
+1='Identify the query string section of a url'
+2='Identify query parameters within a query string'
+3='Explain why an object is a more useful structure for storing query parameters'
```
[the remark this answers is not in the document]
Comment (Dedekind561) [reply content/js2/blocks/query-string/index.md:10]: I just updated to say more than a string

**Comment by Dedekind561** [comment]
> Oh also, do we want to explicitly have an example of "an object containing an array" and "an array containing an object" somewhere, to show composition?

I think the composition of data structures like that should be a major thread in a subsequent week. However, we can certainly include backlog exercises with arrays, including objects etc.

**Comment by Dedekind561** [comment]
@illicitonion Are you happy for me to go ahead and merge this PR?

**Review by illicitonion — APPROVED** [review]
It'd be nice to add the "go research what 'serializes to the same string' means", otherwise LGTM, ship it!

**Pull request merged**